### ISSUE TITLE
eslint returns exit code 8 with emacs client/server setup

### ISSUE THREAD

**Issue by faassen** [eslint/eslint#7353]
I'm trying to make flycheck work with eslint in emacs. After a lot of debugging it _only_ happens when I run emacs as a server and start clients with a emacsclient, not when I directly start emacs. That setup may explain why my problem is not reported commonly.

I found that beyond version flycheck v3.0.0 (even v3.0.1!) flycheck fails to run eslint; v3.0.0 and below work. I have gone a little way in debugging why this may be the case. In the next section therefore I'll talk about eslint versions newer than v3.0.0.

flycheck automatically disables the javascript-eslint checker because it claims the config file is missing. But the eslint config file isn't missing.

Debugging this let me to this bit of elisp code inside of flycheck:

```
(defun flycheck-eslint-config-exists-p ()
  "Whether there is an eslint config for the current buffer."
  (let* ((executable (flycheck-find-checker-executable 'javascript-eslint))
         (exitcode (and executable (call-process executable nil nil nil
                                                 "--print-config" "."))))
    (eq exitcode 0)))
```

So, flycheck basically calls `eslint --print-config .` and checks whether the exit code is 0, and in that case the config file is considered to be present. I've verified that this command works on my command-line, and indeed delivers exit code 0.

But in emacsclient, with the failing versions of eslint, the exit code is 8. With working versions of eslint, the exit code is 0 as I expect.

You can try this in emacs by doing:

`M-:` (pres esc, then `:` key)

You will see `Eval` in the minibuffer.

Then you can type:

`(call-process "eslint" nil nil nil "--print-config" ".")`

I expect to see `0` in the emacs messages buffer (emacs adds some fluff after: `(#o0, #x0, ?\C-@)`). But instead I get `8 (#o10, #x8, ?\C-h)`. I also get an 8 when I just do: `(call-process "eslint")`.

I'm on Linux, my Emacs version is 24.5 and tried it with node v6.8.0 and node v4.3.1.

I did some searching on why eslint or node report the exit code 8. It's not officially documented as a valid node exit code: 

https://github.com/nodejs/node/blob/master/doc/api/process.md#exit-codes

Though this issue may provide a clue, or maybe not:

https://github.com/nodejs/node-v0.x-archive/issues/7677

What could have changed in eslint v3.0.1 to trigger this behavior?

**Comment by not-an-aardvark** [comment]
Thanks for the report.

If you're unable to reproduce the exit code issue on the command line, my guess is that the exit code problem isn't with ESLint; it's something else about your setup that is causing the wrong exit code to be reported. I don't think ESLint's exit code would be different depending on whether it's run from emacs.

To clarify, are you saying that this worked in eslint v3.0.0 and not in eslint v3.0.1? Based on the [list of commits](https://github.com/eslint/eslint/releases/tag/v3.0.1), maybe https://github.com/eslint/eslint/commit/6ef2cbe83b122a835b13e757321972ddb9ab3b9b is the issue (or at least, I'm not sure what other commit it would have been).

**Comment by faassen** [comment]
Yes, it works with eslint v3.0.0 but not with eslint v3.0.1.

I realized that my distinguishing emacs server/client from plain emacs may have been premature -- when I tested with plain emacs I didn't load the rest of my emacs configuration, so that may also have been an issue.

**Comment by faassen** [comment]
I tried removing ``stripBom`` call from the change you indicated but don't see a difference in behavior. Nonetheless, it may be that my quick hack of directly editing the installed file in ``node_modules`` is not the right way to test this.

**Comment by eslintbot** [comment]
Hi @faassen, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by vitorbal** [comment]
Hi @faassen, as the bot says above, would you mind filling in the bug report template? It helps us understand quickly what you are experiencing and what your environment is like, and therefore there's a higher chance someone from the team knows how to help you out. Thanks for understanding!

**Comment by kaicataldo** [comment]
Closing this issue since we haven't heard anything back for quite some time. Feel free to create a new issue if this is something you would like to discuss further.
